cmd/network-interface-manager/main.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2022 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"net"
+	"time"
+
+	"github.com/vishvananda/netlink"
+
+	kubermaticlog "k8c.io/kubermatic/v2/pkg/log"
+)
+
+const (
+	AF_INET    = 0x2
+	AF_INET6   = 0xa
+	SCOPE_HOST = 254
+)
+
+func main() {
+	var (
+		mode   string
+		ifName string
+		ifAddr string
+	)
+
+	var link *netlink.Dummy
+	var err error
+	logOpts := kubermaticlog.NewDefaultOptions()
+	logOpts.AddFlags(flag.CommandLine)
+
+	flag.StringVar(&mode, "mode", "", "Start mode for the process, it can be init or probe")
+	flag.StringVar(&ifName, "if", "", "Name of the network interface to be created or managed, eg: envoyagent")
+	flag.StringVar(&ifAddr, "addr", "", "Network Interface address")
+	flag.Parse()
+
+	rawLog := kubermaticlog.New(logOpts.Debug, logOpts.Format)
+	log := rawLog.Sugar()
+
+	// 'init' mode is for init containers
+	if mode == "init" {
+		link, err = createInterface(ifName)
```

Comment:
we should probably check if the interface does not already exists (and the same for the address)